.github/dependabot.yml

```diff
@@ -14,3 +14,12 @@ updates:
     schedule:
       # Check for updates to GitHub Actions every week
       interval: "weekly"
+
+  # Perform checks and updates for python poetry environment.
+  # See here for more information:
+  # https://docs.github.com/en/code-security/dependabot/dependabot-version-updates/configuration-options-for-the-dependabot.yml-file#pip-and-pip-compile
+  - package-ecosystem: "pip"
+    directory: "/"
+    schedule:
+      # Check for updates to poetry lockfile every week
+      interval: "weekly"
```

Comment:
Thanks good question (this is a bit hard to find in the docs): my understanding is that this will create weekly PR's for each dependency which needs updating (to keep the changes isolated to only those dependencies which are updated). The weekly cadence starts when this gets merged in, so if merged today, weekly on Mondays. Note: there are settings to enable auto-merge in addition to auto-open-PR I believe, but my feeling is that we should have human reviewers take a glance at things and time merges in alignment with other human-focused work (at least for now).